### PR TITLE
Enum values can contain spaces now

### PR DESCRIPTION
**Pull request by mpscholten**
Fixes https://github.com/digitallyinduced/ihp/issues/528

### CODE REVIEW

**Comment by mpscholten** [comment]
Fixes https://github.com/digitallyinduced/ihp/issues/571

File: IHP/NameSupport.hs [hunk 1]
```diff
@@ -57,6 +61,31 @@ tableNameToControllerName tableName = do
...
+        words = 
+                enumValue
+                |> splitOn " "
+                |> List.map (Inflector.parseSnakeCase [])
+                |> List.map (\case
+                        Left failed -> error (cs $ "enumValueToControllerName failed for " <> show failed)
+                        Right result -> result)
+                |> join
+    in
+        Inflector.camelizeCustom True words
```
Comment (nghamilton) [inline IHP/NameSupport.hs:87]: This PR results in camel-case data constructors for enums that are all-caps. As an example, an enum of `'USD'` is now produces a `Usd`, whereas the generated data constructor was previously `USD`. 

Is this the intended behaviour?

**Review by nghamilton — COMMENTED** [review]
@mpscholten Is this expected behaviour of the code gen for all-caps enums, or would you like me to create an issue?

File: IHP/NameSupport.hs (hunk 1 above)
Comment (mpscholten) [reply IHP/NameSupport.hs:87]: Yes, this is expected behavior 👍

**Pull request merged**